tests/JsonSchema/TypeFactoryTest.php

```diff
@@ -35,13 +35,89 @@ public function testGetType(array $schema, Type $type): void
     public function typeProvider(): iterable
     {
         yield [['type' => 'integer'], new Type(Type::BUILTIN_TYPE_INT)];
+        yield [['type' => ['integer', 'null']], new Type(Type::BUILTIN_TYPE_INT, true)];
         yield [['type' => 'number'], new Type(Type::BUILTIN_TYPE_FLOAT)];
+        yield [['type' => ['number', 'null']], new Type(Type::BUILTIN_TYPE_FLOAT, true)];
         yield [['type' => 'boolean'], new Type(Type::BUILTIN_TYPE_BOOL)];
-        yield [['type' => 'string'], new Type(Type::BUILTIN_TYPE_OBJECT)];
+        yield [['type' => ['boolean', 'null']], new Type(Type::BUILTIN_TYPE_BOOL, true)];
+        yield [['type' => 'string'], new Type(Type::BUILTIN_TYPE_STRING)];
+        yield [['type' => ['string', 'null']], new Type(Type::BUILTIN_TYPE_STRING, true)];
+        yield [['type' => 'object'], new Type(Type::BUILTIN_TYPE_OBJECT)];
+        yield [['type' => ['object', 'null']], new Type(Type::BUILTIN_TYPE_OBJECT, true)];
         yield [['type' => 'string', 'format' => 'date-time'], new Type(Type::BUILTIN_TYPE_OBJECT, false, \DateTimeImmutable::class)];
+        yield [['type' => ['string', 'null'], 'format' => 'date-time'], new Type(Type::BUILTIN_TYPE_OBJECT, true, \DateTimeImmutable::class)];
         yield [['type' => 'string', 'format' => 'duration'], new Type(Type::BUILTIN_TYPE_OBJECT, false, \DateInterval::class)];
-        yield [['type' => 'string'], new Type(Type::BUILTIN_TYPE_OBJECT, false, Dummy::class)];
+        yield [['type' => 'object'], new Type(Type::BUILTIN_TYPE_OBJECT, false, Dummy::class)];
```

Comment:
Please mind this change: I don't understand why `object` was being mapped as `string` previously, but I suppose the idea was to map UUID-alike single-field value types? That is indeed wrong, unless I'm missing a basic use-case.

Reply:
It's because an object is normalized as a string identifier ie `/users/1` (I guess?).